### PR TITLE
Start permission integration

### PR DESCRIPTION
**Pull request by BrianSilvia**
Permissions should be good, just need to resolve the mongoose model issue in the test (cannot overwrite model once compiled)

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -66,27 +48,135 @@ describe( 'action', function() {
...
+            // const permission2 = new Permission({
+            //     resourceType: 'file',
+            //     resourceId: '614dd78b-0d7b-4777-8aa6-c9d20dfb3032',
+            //     userId: 'userId',
+            //     read: true,
+            //     write: true,
+            //     destroy: true,
+            // });
+            // permission2.save();
+
```
Comment (JustinLivi) [inline test/index.js]: All these saves are asynchronous and we should wait for them to finish with a `Promise.all`.
We can use the `create` shortcut function and then inline the promise into an array:

```
Promise.all([
    File.create({
        // options for file 1
    }),
    File.create({
        // options for file 2
    }),
]);
```

File: test/index.js [hunk 2]
```diff
@@ -174,16 +281,13 @@ describe( 'action', function() {
...
-                    dateCreated: 1439765335,
-                    lastModified: 1439765353,
-                })
-            )
+            .expect( res => {
+                expect( res.body ).to.have.property( '_id', '614dd78b-0d7b-4777-8aa6-c9d20dfb3032' );
+                expect( res.body ).to.have.property( 'dateCreated', '2016-06-19T20:29:57.000Z' );
+                expect( res.body ).to.have.property( 'lastModified', '2016-06-19T20:29:57.000Z' );
+                expect( res.body ).to.have.property( 'mimeType', 'image/gif' );
+                // expect( res.body ).to.have.property( 'parents', [ '8d461483-e701-47ca-8788-c6210550fdb9' ]);
```
Comment (JustinLivi) [inline test/index.js]: Does this not work?

File: test/index.js [hunk 3]
```diff
@@ -231,14 +330,47 @@ describe( 'action', function() {
...
             });
         });
     });
 
-    describe( 'put', function() {});
-    describe( 'delete', function() {});
+    describe( 'put', function() {
+        it( 'should update both the s3 and the mongodb document', function() {
+            return new Promise(( resolve, reject ) => {
+                request.post({
```
Comment (JustinLivi) [inline test/index.js]: We're describing 'put' but we're making a post request. Which is it supposed to be?

File: test/index.js (hunk 2 above)
Comment (BrianSilvia) [inline test/index.js]: the created and modified times are generated on the server, which causes issues

**Pull request merged**